### DISCUSSION TITLE
Can't disable pre-allocation on Mac

### DISCUSSION THREAD

**Discussion by vasyl-shumskyi** [transmission/transmission#1958] [Q&A]
I tried to disable Transmission pre-allocation on Mac to prevent wearing out SSD lifespan. 

To do so I run the following command in the Terminal as mentioned in [Help Docs](https://github.com/transmission/transmission/wiki/Editing-Configuration-Files)

```sh
$ defaults write org.m0k.transmission preallocation 0
```

I have made sure that the option set correctly with the following commands:

```sh
$ cp  $HOME/Library/Preferences/org.m0k.transmission.plist transmission.bin.plist
$ plutil -convert xml1 transmission.bin.plist
$ cat transmission.bin.plist
```

```xml
....
<key>preallocation</key>
<string>0</string>
```
I have also tried to set this options with `:` without luck:

```sh
$ defaults write org.m0k.transmission preallocation: 0
```

After restarting the Transmission app it still preallocates the file before downloading it. 

Did I miss something? Do I need to run Transmission daemon to make this work? If so, how do I do that? 

Another question I have, does fast preallocation which is set as default 1 wear out SSD? 
Or it only writes files header to reserve file size on disk and nothing more?  

Also some torrents may stuck forever in downloading because there are no seeders for weeks, so it would be wise not to pre-allocate full file size to save disk space from such idle torrents.

It would be awesome if such a valuable option present in client Settings so users can change it via GUI. 

Thanks!

**Comment by parasyte** [comment]
Why was this transferred from an issue to a discussion? This is a bug that I hit. Setting the `preallocation` key to 0 has no effect.

I'm using NFS, and file preallocation blocks the UI thread (see #2462) for upwards of 15 minutes for large files, like a typical UHD movie.
